analysis/hunterbeastmastery/src/modules/apl/AplCheck.tsx

```diff
@@ -0,0 +1,108 @@
+import SPELLS from 'common/SPELLS';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { WIPSuggestionFactory } from 'parser/core/CombatLogParser';
+import aplCheck, { build } from 'parser/shared/metrics/apl';
+import annotateTimeline from 'parser/shared/metrics/apl/annotate';
+import {
+  buffPresent,
+  buffMissing,
+  hasConduit,
+  or,
+  inExecute,
+  and,
+  spellCharges,
+  not,
+  spellAvailable,
+  hasResource,
+  hasLegendary,
+} from 'parser/shared/metrics/apl/conditions';
+
+export const apl = build([
+  {
+    spell: SPELLS.BARBED_SHOT,
+    condition: buffMissing(SPELLS.BARBED_SHOT_PET_BUFF, { timeRemaining: 1000, duration: 8000 }),
+  },
+  {
+    spell: SPELLS.BARBED_SHOT,
+    condition: and(
+      buffPresent(SPELLS.WILD_SPIRITS_BUFF),
+      hasLegendary(SPELLS.FRAGMENTS_OF_THE_ELDER_ANTLERS),
+    ),
+  },
+  {
+    spell: SPELLS.TAR_TRAP,
+    condition: and(hasLegendary(SPELLS.SOULFORGE_EMBERS_EFFECT), spellAvailable(SPELLS.FLARE)),
+  },
+  {
+    spell: SPELLS.FLARE,
+    condition: and(
+      hasLegendary(SPELLS.SOULFORGE_EMBERS_EFFECT),
+      not(spellAvailable(SPELLS.TAR_TRAP)),
+    ),
+  },
+  SPELLS.BLOODSHED_TALENT,
+  SPELLS.WILD_SPIRITS,
+  SPELLS.FLAYED_SHOT,
+  {
+    spell: SPELLS.KILL_SHOT_MM_BM,
+    condition: inExecute(0.2),
+  },
+  {
+    spell: SPELLS.KILL_SHOT_MM_BM,
+    condition: buffPresent(SPELLS.FLAYERS_MARK),
+  },
+  SPELLS.WAILING_ARROW_CAST,
+  {
+    spell: SPELLS.DEATH_CHAKRAM_INITIAL_AND_AOE,
+    condition: hasResource(RESOURCE_TYPES.FOCUS, { atMost: 80 }),
```

Comment:
as discussed in discord, this *probably* doesn't work as intended yet because `hasResource` relies solely on `resourcechange` events.

Reply:
should work now